tern/classes/package.py

```diff
@@ -90,11 +100,33 @@ def checksum(self):
     def checksum(self, checksum):
         self.__checksum = checksum
 
+    def get_file_paths(self):
+        """Return a list of paths of all the files in a package"""
+        return list(map(lambda f: f.path, self.__files))
+
+    def add_file(self, file):
+        """Add a file to the list of files present in a package"""
+        if isinstance(file, FileData):
+            if file.path not in self.get_file_paths():
```

Comment:
I don't think I added this extra check in the `add_file` function in the `ImageLayer` class. Nice!